lib/Mojo/Util.pm

```diff
@@ -130,6 +153,17 @@ sub dumper { Data::Dumper->new([@_])->Indent(1)->Sortkeys(1)->Terse(1)->Useqq(1)
 
 sub encode { _encoding($_[0])->encode("$_[1]", 0) }
 
+sub encrypt_cookie {
+  my ($value, $key, $salt) = @_;
+  croak 'CryptX 0.080+ required for encrypted cookie support' unless CRYPTX;
+
+  my $dk = Crypt::KeyDerivation::pbkdf2($key, $salt);
```

Comment:
The expanded key only has to be computed once at application start up. Running pbkdf2 each time a cookie is encrypted or decrypted is computationally expensive and will tank Mojo performance.

This is the same performance problem as in mojo.js, as discussed in https://github.com/mojolicious/mojo/pull/2200#issuecomment-2407646319. However, mojo.js uses the more modern (and more computationally expensive) scrypt key-derivation function.

It could also be sensible to tune the PBKDF2 cost parameter. The `Crypt::KeyDerivation` documentation says that it defaults to 5000 iterations, whereas 2023-era OWASP guidelines suggest 600,000 iterations for password storage purposes. However, a low cost parameter may actually be appropriate here.

Reply:
It's now cached.